Post-shader error: 0(2): error C0121: #version 300 must be followed by es...

### What happens?
Following upgrading from 1.5.4 to 1.6.3.  When using Rendering Backend->OpenGL with any postprocessing shader selected (ie FXAA Antialiasing) upon loading a game (any) get an error message on screen (see issue title) and the shader is apparently NOT active.  After far too much hunting around found (from source code) how to generate a log file, got the following snippet:
`
59:14:677 idle0        E[G3D]: thin3d\GLQueueRunner.cpp:223 Error in shader compilation for: postshader
59:14:677 idle0        E[G3D]: thin3d\GLQueueRunner.cpp:224 Info log: 0(2) : error C0121: #version 300 must be followed by es
0(2) : error C0201: unsupported version 300
0(6) : error C7532: global type vec2 requires "#version 100" or later
0(6) : error C5060: out can't be used with non-varying v_texcoord0
0(8) : error C7532: global type vec4 requires "#version 100" or later
0(13) : error C7532: global variable gl_Position requires "#version 100" or later

59:14:678 idle0        E[G3D]: thin3d\GLQueueRunner.cpp:225 Shader source:
#version 300
#ifdef GL_ARB_shading_language_420pack
#extension GL_ARB_shading_language_420pack : require
#endif

out vec2 v_texcoord0;
in vec2 a_texcoord0;
in vec4 a_position;

void main()
{
    v_texcoord0 = a_texcoord0;
    gl_Position = a_position;
    gl_Position.z = 2.0 * gl_Position.z - gl_Position.w;
}
`

There are many other snippets with a similar initial error but I don't know how one would post the log to github as the help suggested doing (btw nifty new issue preset).

### What *should* happen?
As in 1.5.4 with opengl backend, game loads and shader appears to be active with no errors.

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.
Desktop Windows 7 GeForce DX10 (not 10.1) graphics card.

### Troubleshooting:
-Tried DX9, no shaders available so not applicable.  Vulkan is (as expected) not an option.
-Tried DX11 and surprisingly it worked and the shaders appeared to work (tried a few different ones and they did what they claimed).

In case there were any left over old/invalid files lingering around I also tried a fresh install and got the same error, after I ensured that backend was opengl and a postprocessing shader was selected that is.  To put it another way this is happening with defaults + 2 changes.

--- a/GPU/Common/ShaderTranslation.cpp
+++ b/GPU/Common/ShaderTranslation.cpp
@@ -310,11 +310,7 @@ bool TranslateShader(std::string *dest, ShaderLanguage destLang, TranslatedShade
 		spirv_cross::ShaderResources resources = glsl.get_shader_resources();
 		// Set some options.
 		spirv_cross::CompilerGLSL::Options options;
-		if (gl_extensions.ver[0] >= 4) {
-			options.version = 400;
-		} else {
-			options.version = 300;
-		}
+		options.version = gl_extensions.GLSLVersion();
 		glsl.set_options(options);
 		// Compile to GLSL, ready to give to GL driver.
 		*dest = glsl.compile();

--- a/ext/native/gfx_es2/gpu_features.cpp
+++ b/ext/native/gfx_es2/gpu_features.cpp
@@ -54,6 +54,23 @@ bool GLExtensions::VersionGEThan(int major, int minor, int sub) {
 	return gl_extensions.ver[2] >= sub;
 }
 
+int GLExtensions::GLSLVersion() {
+	// Used for shader translation and core contexts (Apple drives fail without an exact match.)
+	if (gl_extensions.VersionGEThan(3, 3)) {
+		return gl_extensions.ver[0] * 100 + gl_extensions.ver[1] * 10;
+	} else if (gl_extensions.VersionGEThan(3, 2)) {
+		return 150;
+	} else if (gl_extensions.VersionGEThan(3, 1)) {
+		return 140;
+	} else if (gl_extensions.VersionGEThan(3, 0)) {
+		return 130;
+	} else if (gl_extensions.VersionGEThan(2, 1)) {
+		return 120;
+	} else {
+		return 110;
+	}
+}
+
 void ProcessGPUFeatures() {
 	gl_extensions.bugs = 0;
 
@@ -505,15 +522,7 @@ std::string ApplyGLSLPrelude(const std::string &source, uint32_t stage) {
 	std::string version = "";
 	if (!gl_extensions.IsGLES && gl_extensions.IsCoreContext) {
 		// We need to add a corresponding #version.  Apple drives fail without an exact match.
-		if (gl_extensions.VersionGEThan(3, 3)) {
-			version = StringFromFormat("#version %d%d0\n", gl_extensions.ver[0], gl_extensions.ver[1]);
-		} else if (gl_extensions.VersionGEThan(3, 2)) {
-			version = "#version 150\n";
-		} else if (gl_extensions.VersionGEThan(3, 1)) {
-			version = "#version 140\n";
-		} else {
-			version = "#version 130\n";
-		}
+		version = StringFromFormat("#version %d\n", gl_extensions.GLSLVersion());
 	}
 	if (stage == GL_FRAGMENT_SHADER) {
 		temp = version + glsl_fragment_prelude + source;

--- a/ext/native/gfx_es2/gpu_features.h
+++ b/ext/native/gfx_es2/gpu_features.h
@@ -103,6 +103,7 @@ struct GLExtensions {
 
 	// greater-or-equal than
 	bool VersionGEThan(int major, int minor, int sub = 0);
+	int GLSLVersion();
 };
 
 extern GLExtensions gl_extensions;

